File: backend/src/main/java/com/project/isa/model/Promotion.java
package com.project.isa.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;


@Data
@Entity
@Table(name = "PROMOTION")
public class Promotion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date startPromo;

    private Date endPromo;

    private String description;

    @ManyToOne
    private Adventure adventurePromotion;

    @ManyToOne
    private Boat boatPromotion;

    @ManyToOne
    private VacationHome vacationHomePromotion;

}

File: frontend/src/app/model/promotion.ts
import { Adventure } from "./adventure";
import { Boat } from "./boat";
import { VacationHome } from "./vacation-home";

export class Promotion {

    id: string;
    startPromo: string;
    endPromo: string;
    description: string;
    adventurePromotion: Adventure;
    boatPromotion: Boat;
    vacationHomePromotion: VacationHome;
    numberOfPromotions: number;
}
